frontend/app/player/MessageDistributor/managers/ListWalker.ts

```diff
@@ -118,4 +118,19 @@ export default class ListWalker<T extends Timed> {
 		}
 	}
 
+	async moveWait(t: number, fn: (msg: T) => Promise<any> | undefined, fnBack?: (msg: T) => void): Promise<void> {
+		// Applying only in increment order for now
+		if (t < this.timeNow) {
+			this.reset();
+		}
+
+		while (!!this._list[this.p] && this._list[this.p].time <= t) {
```

Comment:
this line is a bit hard to understand, can be simplified by creating named vars (that describe themselves)